pkgs/native_assets_cli/lib/src/api/build_mode.dart

```diff
@@ -2,14 +2,36 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import '../model/build_mode.dart' as model;
+part '../model/build_mode.dart';
 
-abstract class BuildMode {
+/// The build mode for compiling native code assets.
+///
+/// The Dart SDK does not have build modes. All `build.dart` invocations
+/// are invoked with [release].
+///
+/// The Flutter SDK build modes map as the following to `build.dart` build
+/// modes:
+/// * Flutter debug -> [debug].
+/// * Flutter release -> [release].
+/// * Flutter profile -> [release].
+/// * Flutter jit release -> [release].
+abstract final class BuildMode {
+  /// The name for this build mode.
   String get name;
 
-  static const BuildMode debug = model.BuildMode.debug;
-  static const BuildMode release = model.BuildMode.release;
+  /// The debug build mode.
+  ///
+  /// Used by the Flutter SDK in its debug mode.
+  static const BuildMode debug = BuildModeImpl.debug;
 
+  /// The release build mode.
+  ///
+  /// Used by the Flutter SDK in its release, profile, and jit release modes.
+  ///
+  /// Used by the Dart SDK for every build.
+  static const BuildMode release = BuildModeImpl.release;
+
+  /// All known build modes.
   static const values = <BuildMode>[
```

Comment:
Is this needed?

Reply:
I was modeling this kind of as an open ended enum. E.g. we can add values in the future, but users can iterate over all known values.

We're for example iterating over the known values in the unit tests of `package:native_toolchain_c` to build both static and dynamic libraries.

I can be pursuaded to remove this if you think that's better.

Reply:
No strong opinion, but I think it would make a tad cleaner to not have the `values` field. It doesn't feel like something which needs iterating over... It probably boils down to if adding new `BuildMode`s should automatically be tested in `native_toolchain_c` or other uses of `values`.

Reply:
Let's keep it for now then. A lot of other classes in this repo are organized the same way with a `values` static field.